### PR TITLE
feature: Adapt the SmartDocuments integration documentation

### PR DESCRIPTION
**Pull request by hsiliev**
Remove DMS plugin section
Add redirect/callback steps
Add configuration section

Solves: PZ-3732

### CODE REVIEW

File: docs/solution-architecture/smartDocumentsIntegration.md [hunk 1]
```diff
@@ -1,97 +1,54 @@
-# SmartDocuments document creation flow
+# SmartDocuments
+
+## Configuration
```
Comment (bas-info-nl) [inline docs/solution-architecture/smartDocumentsIntegration.md:3]:
```suggestion
## Case Configurations 
```

File: docs/solution-architecture/smartDocumentsIntegration.md [hunk 2]
```diff
@@ -1,97 +1,54 @@
-# SmartDocuments document creation flow
+# SmartDocuments
+
+## Configuration 
+
+The configuration of every case type contains a mapping between SmartDocuments template and document type. This mapping is used to get the correct template and template group with the Document creation flow below.
```
Comment (bas-info-nl) [inline docs/solution-architecture/smartDocumentsIntegration.md:5]:
```suggestion
Each case type configuration defines a mapping between SmartDocuments template and document type. This mapping is used to get the correct template and template group with the Document creation flow below.
```

Comment (bas-info-nl) [reply docs/solution-architecture/smartDocumentsIntegration.md:5]: And should we add something on the document 'confidentiality' level?

File: docs/solution-architecture/smartDocumentsIntegration.md [hunk 3]
```diff
@@ -1,97 +1,54 @@
...
-8. When the user returns to the ZAC browser tab, they will see the new document attached to the case or task.
+3. The document creation wizard is started in SmartDocuments. Case or task information and template type are pre-filled.
+4. At the end of the document creation wizard, SmartDocuments:
+   1. creates a Word document and stores it on SmartDocuments side for later retrieval
+   2. calls a ZAC callback endpoint with information about the created document
+5. ZAC callback endpoint in turn:
+   1. Downloads the Word document from SmartDocuments 
+   2. Creates a document ('enkelvoudiginformatieobject') in the ZRC API implementation (also known as 'zaakregister'; in the context of ZAC this is usually OpenZaak).
+   3. Usage permissions ('Gebruiksrechten') are set for the document. Without this step the zaak to which this document belongs cannot be closed nor interrupted.  
+   4. The document/enkelvoudiginformatieobject is linked to the case or task in question (by creating a ‘zaakinformatieobject’).
```
Comment (bas-info-nl) [inline docs/solution-architecture/smartDocumentsIntegration.md:21]: Add something on 'confidentiality' level applied to the document?

**Review by bas-info-nl — COMMENTED** [review]
Just a couple of wordings and a question on the 'confidentiality' level that is configured and applied to the document.

File: docs/solution-architecture/smartDocumentsIntegration.md (hunk 3 above)
Comment (hsiliev) [reply docs/solution-architecture/smartDocumentsIntegration.md]: Added:
`3. Confidentiality of the document is set to PUBLIC and the document status to IN_PROGRESS`

File: docs/solution-architecture/smartDocumentsIntegration.md [hunk 4]
```diff
@@ -1,97 +1,55 @@
...
+Employee->>SmartDocuments: Fill in remaining document data
+Employee->>SmartDocuments: Click "Finish" in the creation wizard
+SmartDocuments->>SmartDocuments: Store document
+SmartDocuments--)+ZAC: Provide document information
+Employee->>SmartDocuments: Close document creation wizard
+
+ZAC->>SmartDocuments: Download document
+ZAC->>+OpenZaak: Link document to case or task
+OpenZaak->>-OpenNotificaties: Send notification that a new document is linked to a case or task
+OpenNotificaties--)ZAC: Recived notification "document linked to a case"
```
Comment (infonl-marcel) [inline docs/solution-architecture/smartDocumentsIntegration.md:53]: Typo: Received

**Review by infonl-marcel — APPROVED** [review]
Great Dutch texts @hsiliev ! 😉

File: docs/solution-architecture/smartDocumentsIntegration.md (hunk 4 above)
Comment (hsiliev) [reply docs/solution-architecture/smartDocumentsIntegration.md]: Fixed! Thanks for noticing this!

**Pull request merged**